projects/src/main/java/org/odk/collect/projects/AddProjectDialog.kt

```diff
@@ -39,17 +40,28 @@ class AddProjectDialog : MaterialFullScreenDialogFragment() {
         super.onViewCreated(view, savedInstanceState)
         setUpToolbar()
 
-        binding.projectIconInputText.addTextChangedListener(OneSignTextWatcher(binding.projectIconInputText))
+        binding.urlInputText.doOnTextChanged { text, _, _, _ ->
+            if (Validator.isUrlValid(text.toString())) {
+                binding.urlInputText.error = null
+                binding.addButton.isEnabled = true
+                binding.addButton.alpha = 1f
+            } else {
+                binding.urlInputText.error = getString(R.string.url_error)
+                binding.addButton.isEnabled = false
+                binding.addButton.alpha = 0.3f
```

Comment:
I tried to do that using styling but somehow wasn't able. I think there might be something wrong with it but @seadowg if you want to play with it a bit you can do that.